src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -2914,6 +2753,956 @@ void QgsOgrProvider::forceReload()
+      break;
+    case wkbMultiSurfaceZM:
+      geom = "MultiSurfaceZM";
+      break;
+# else
+      // Note: [build with gdal 2.*] when running with gdal 1.*, these values are not returned - thus not queried -no failures
+      // Note: [build with gdal 1.*] when running with gdal 2.*, these values are returned - but unknown al build time -no failures
+    case 2001:
+      geom = "PointM";
+      break;
+    case 2002:
+      geom = "LineStringM";
+      break;
+    case 2003:
+      geom = "PolygonM";
+      break;
+    case 2004:
+      geom = "MultiPointM";
+      break;
+    case 2005:
+      geom = "MultiLineStringM";
+      break;
+    case 2006:
+      geom = "MultiPolygonM";
+      break;
+    case 2007:
+      geom = "GeometryCollectionM";
+      break;
+    case 3001:
+      geom = "PointZM";
+      break;
+    case 3002:
+      geom = "LineStringZM";
+      break;
+    case 3003:
+      geom = "PolygonZM";
+      break;
+    case 3004:
+      geom = "MultiPointZM";
+      break;
+    case 3005:
+      geom = "MultiLineStringZM";
+      break;
+    case 3006:
+      geom = "MultiPolygonZM";
+      break;
+    case 3007:
+      geom = "GeometryCollectionZM";
+      break;
+    case 2008:
+      geom = "CircularStringM";
+      break;
+    case 2009:
+      geom = "CompoundCurveM";
+      break;
+    case 2010:
+      geom = "CurvePolygonM";
+      break;
+    case 2011:
+      geom = "MultiCurveM";
+      break;
+    case 2012:
+      geom = "MultiSurfaceM";
+      break;
+    case 3008:
+      geom = "CircularStringZM";
+      break;
+    case 3009:
+      geom = "CompoundCurveZM";
+      break;
+    case 3010:
+      geom = "CurvePolygonZM";
+      break;
+    case 3011:
+      geom = "MultiCurveZM";
+      break;
+    case 3012:
+      geom = "MultiSurfaceZM";
+      break;
+#endif
+#if defined(GDAL_COMPUTE_VERSION) && GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(2,1,0)
+      // ISO SQL/MM Part 3. GDAL &gt;= 2.1
+    case wkbCurve:
+    case wkbSurface:
+    case wkbCurveZ:
+    case wkbSurfaceZ:
+    case wkbCurveM:
+    case wkbSurfaceM:
+    case wkbCurveZM:
+    case wkbSurfaceZM:
+      // Reserved in GDAL &gt;= 2.1 but not yet implemented
+    case wkbPolyhedralSurface:
+    case wkbTIN:
+    case wkbTriangle:
+    case wkbPolyhedralSurfaceZ:
+    case wkbTINZ:
+    case wkbTriangleZ:
+    case wkbPolyhedralSurfaceM:
+    case wkbTINM:
+    case wkbTriangleM:
+    case wkbPolyhedralSurfaceZM:
+    case wkbTINZM:
+    case wkbTriangleZM:
+#endif
+    default:
+      // Do not use ':', as it will mess with the separator used by QgsSublayersDialog::populateLayers()
+      geom = QString( "Unknown WKB (%1)" ).arg( type );
+  }
+  return geom;
+}
+
+QgsWKBTypes::Type QgsOgrProviderUtils::wkbGeometryTypeFromName( const QString& typeName )
+{ // this should be keep uptodate with gdal/ogr/ogr_core.h (as returned in QgsOgrProvider::ogrWkbGeometryTypeName)
+  if ( typeName == "Point" ) return QgsWKBTypes::Point;
```

Comment:
This was initially intended to replace the OGR types checking
- which caused problems with the different gdal versions (symbol lookup error)
   - so the first goal was to get it working with QgsWkbTypes

While doing so I was indeed wondering if 
- QgsOgrProviderUtils::wkbGeometryTypeName
    - could first call QgsWKBTypes::parseType
and
- QgsOgrProviderUtils::wkbGeometryTypeFromName
    - could first call QgsWKBTypes::displayString( Type type )

and then filter out any types that are known to QGIS but not known to OGR
- there are some types defined but not implemented and thus must be filtered out

I will try this out tomorrow and see if I can optimize this part.